packages/snowpack/src/build/build-pipeline.ts

```diff
@@ -111,6 +111,34 @@ async function runPipelineTransformStep(
   return output;
 }
 
+export function runPipelineProxyStep(
+  fileUrl: string,
+  contents: string,
+  {buildPipeline, messageBus, isDev}: BuildFileOptions,
+) {
+  for (const step of buildPipeline) {
+    if (!step.proxy) {
+      continue;
+    }
+    const result = step.proxy({
+      contents,
+      fileUrl,
+      isDev,
+      log: (msg, data = {}) => {
```

Comment:
note for future usage: it’d be nice to type this out, as I’m not sure of all the ways this can be used